composer/datasets/in_context_learning_evaluation.py

```diff
@@ -912,240 +1106,172 @@ class InContextLearningCodeEvalDataset(Dataset):
-        tokenizer (Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast]): The tokenizer used to map between strings and token ids
-        batch_size (int): Size of a batch used for eval
-        max_seq_len (int): The maximum sequence length supported by the model
-        pad_tok_id (int): The special token reserved for padding batches
-        num_fewshot (int): The number of complete fewshot examples to prepend before each test example
-        prompt_string (str): Prompt string to put once before all fewshot examples/test examples (e.g. 'translate english to french')
-        example_delimiter (str): Separator that goes between individual (context, answer) pairs (e.g. '\n')
-        destination_path (str): Temporary path to store downloaded datasets
-        code_prelimiter (str): String to put before each code prompt (e.g. 'Q: ')
-        fewshot_random_seed (int): Random seed to use for fewshot sampling
-        generations_per_sample: how many outputs to generate per prompt
-        top_p: top_p sampling parameter for nucleus sampling
-        top_k: top_k sampling parameter for number of samples to consider
+
+    Each batch then consists of the following the structure
+    - input_ids: Input tensor batch x seqlen x num tokens
+    - mode: Indicates to the model that this is an ICL task and may rely on a custom code path to properly update metrics
+    - mode: always set to 'generate'
+    - labels: exact solution for the coding problem
+    - prompts: prompt for the task
+    - entry_points: list of entry points
+    - test_inputs: list of test inputs
+    - test_outputs: list of test outputs
+    - languages:  list of languages
+    - pass_at_k: passed value for pass_at_k
+    - generation_length: derrived maximum generation length
+    - generation_kwargs: Dictionary of kwargs neeeded for generation. Includes the following:
+        - pad_token_id: ID for padding token, derived automatically
+        - num_beams: how many beams to search for generations, always set to 1
+        - num_return_sequences: value passed for 'generations_per_sample', how many generations per prompt
+        - do_sample: determines whether model is sampling or greedily decoding. Always set to True
+        - top_p: the cumulative probability of parameter highest probability vocabulary tokens to keep for nucleus sampling. Must be between 0 and 1
+        - top_k: the number of highest probability vocabulary tokens to keep for top-k-filtering. Between 1 and infinity.
+        - temperature: randomness used during prediction. 1.0 is deterministic. defaults to 1.0
+        - use_cache: Whether or not to use past key values to speed up sampling. Always set to True
+
+    Additional Args:
+        # TODO: are these correct?
+        generations_per_sample (int) (defaults to 1): how many outputs to generate per prompt
```

Comment:
Used to provide stable estimates pass@k, I believe it is required that n > k

@mcarbin can you take a look at the way Max is describing these?

Reply:
Found direct definitions in HF generate and have used those definitions.